### DISCUSSION TITLE
Signature interface

### DISCUSSION THREAD

**Discussion by jesusgp22** [directus/directus#7496] [Feature Requests]
I'm trying to build a custom interface that allows the user to input a signature using touch/pen/mouse so here are my thoughts:
1 - Try to convert your already existing file image field into an extension as a base for my work (currently I'm here)
2 - Integrate with an external library https://github.com/szimek/signature_pad 
3 - Rasterize the image and save as Image

What I aim to have is both a working extension and detailed steps and explanation of how to develop extensions, work on top of existing directus components, integrate with external libraries and build.

for 1. I'm having several issues as the directus code uses ts, sass and imports things that extensions can't, I just removed all the typescript and style tag and now trying to figure out the imports

```js
import { useI18n } from 'vue-i18n';
import { defineComponent, ref, watch, computed } from 'vue';
import api from '/api';
import formatFilesize from '@/utils/format-filesize';
import FileLightbox from '@/views/private/components/file-lightbox';
import { nanoid } from 'nanoid';
import { getRootPath } from '@/utils/get-root-path';
import { unexpectedError } from '@/utils/unexpected-error';
import { addTokenToURL } from '@/api';
import DrawerItem from '@/views/private/components/drawer-item';
```

Things like api and addTokenToUrl seem the be available via system injectio, to make the vue and nanoid import work required to install the packages on my package.json (not sure if that will bloat the extension with things already available in the app) 
And I'm only missing importing FileLightbox and DrawerItem and the utils not sure what is the right way to replace the @ import

**Comment by rijkvanzanten** [comment]
Heya!


> Try to convert your already existing file image field into an extension as a base for my work

What image field? What parts are you trying to lift over?

It might be easier to start from scratch, and lift over bits you can use, instead of trying to copy the whole thing, and then remove bits you don't need.

> for 1. I'm having several issues as the directus code uses ts, sass and imports things that extensions can't

Not necessarily, extensions are external codebases that can theoretically do whatever they want, as long as it's being bundled to a final dist that can be read by the app. This includes TypeScript and Sass.

As for the imports, we've started to export more and more of the apps internal utilities through the shared and extensions packages. We can't just expose everything within the app codebase, as that would mean that effectively every tiny change in the app can potentially cause a breaking change in your extensions.



cc. @nickrum

**Comment by jesusgp22** [comment]
Hi @rijkvanzanten! Thank you for your quick reply, 

This is the image field I'm using: https://github.com/directus/directus/blob/main/app/src/interfaces/file-image/file-image.vue I'll try to change my approach 

How can I access or read docs on the available internal utilities and shared packages?

**Reply by rijkvanzanten** [reply]
The `@directus/shared` package exports all the util functions that are meant for use outside of the internals of the codebase: https://github.com/directus/directus/tree/main/packages/shared/src/utils

**Reply by jesusgp22** [reply]
> The `@directus/shared` package exports all the util functions that are meant for use outside of the internals of the codebase: https://github.com/directus/directus/tree/main/packages/shared/src/utils

The utils I needed were not exported from this lib so I ended up looking at the directus source and doing a lot of copy/paste but it worked!

**Comment by jesusgp22** [comment]
I finally have a working extension and it's being battle tested on a production site! Now, what is the right place to contribute with public extensions, tutorials?

**Reply by DeltaCorporate** [reply]
i need a custom field like this can you please share your code?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
